Wgrib2 without ip2

kgerheiser opened the pull request.

kgerheiser committed:
Update COMPONENTS

kgerheiser committed:
Remove wgrib2 dependence on ip2

aerorahul commented:
why remove ip2 from the stack, just remove wgrib2 dependency on ip2.

kgerheiser commented:
What about the flat installation issue?

aerorahul commented:
> What about the flat installation issue?

yeah, you are right. we have to remove it until ip and ip2 are reconciled.
Did you get a chance to see what it will take to do the reconciliation?

aerorahul commented on CMakeLists.txt:
```diff
@@ -98,7 +98,7 @@ endfunction()
 
 # Collect list of components to append to CMAKE_PREFIX_PATH
 list(APPEND _comps
-  bacio sigio sfcio gfsio ip sp w3nco g2 g2tmpl nemsio nemsiogfs landsfcutil w3emc wrf_io bufr ip2 wgrib2)
```
Instead of removing it completely, can you put logic in if (FLAT) then remove, other wise build?

aerorahul replied:
you can also put big bold warning why you are removing it.

aerorahul commented on CMakeLists.txt:
```diff
@@ -120,8 +120,6 @@ string(REPLACE ";" "|" CMAKE_PREFIX_PATH_ALT_SEP "${_cpath}")
 set(nemsio_depends bacio w3nco)
 set(nemsiogfs_depends nemsio)
 set(w3emc_depends nemsio sigio)
-set(ip2_depends sp)
```
this is true and should stay.

aerorahul commented on CMakeLists.txt:
```diff
@@ -120,8 +120,6 @@ string(REPLACE ";" "|" CMAKE_PREFIX_PATH_ALT_SEP "${_cpath}")
 set(nemsiogfs_depends nemsio)
 set(w3emc_depends nemsio sigio)
-set(ip2_depends sp)
-set(wgrib2_depends ip2 sp)
```
Add this line if not installing in FLAT structure.

aerorahul commented on COMPONENTS:
```diff
@@ -10,7 +10,6 @@ nemsiogfs   | v2.5.0            | 2.5.0
 w3emc       | v2.7.0            | 2.7.0
 w3nco       | v2.4.0            | 2.4.0
 ip          | v3.3.0            | 3.3.0
-ip2         | v1.1.0            | 1.1.0
```
keep.

kgerheiser commented:
Ok, I'll do that

kgerheiser committed:
Re-add ip2

If flat structure is ON don't build ip2 because it overwrites ip modules

If not building ip2 build wgrib2 without it. If it is being built build wgrib2 with ip2 support.

kgerheiser committed:
Re-add ip2 to COMPONENTS

kgerheiser committed:
Fix typo

kgerheiser commented:
Fixed

aerorahul reviewed (approved):
looks good.
thanks for the fixes.
